presto-spi/src/main/java/com/facebook/presto/spi/statistics/Range.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.statistics;
+
+import java.util.Objects;
+
+import static java.lang.Double.isNaN;
+import static java.lang.String.format;
+
+public class Range
+{
+    private final double min;
+    private final double max;
+
+    public Range(double min, double max)
+    {
+        if (isNaN(min)) {
+            throw new IllegalArgumentException("min must not be NaN");
+        }
+        if (isNaN(max)) {
+            throw new IllegalArgumentException("max must not be NaN");
+        }
+        if (max < min) {
+            throw new IllegalArgumentException(format("max must be greater than or equal to min. max: %s. min: %s.", max, min));
```

Comment:
nit: express in ctor params order:
```
if (min > max) {
            throw new IllegalArgumentException(format("min must be lower than or equal to max. min: %s. max: %s.", min, max));
```

Reply:
Or even simpler
```java
format("min (%s) cannot be larger than max (%s)", min, max)

Reply:
@electrum I don't like to mix the values with the static part of an error message, as it is harder than to search for it in the codebase. If you don't mind I would like to leave it as is.